packages/multichain/src/scope/supported.ts

```diff
@@ -0,0 +1,138 @@
+ * @param account - The CAIP account ID to check.
+ * @param getInternalAccounts - A function that returns the internal accounts.
+ * @returns A boolean indicating if the account is supported by the wallet.
+ */
+export const isSupportedAccount = (
+  account: CaipAccountId,
+  getInternalAccounts: () => { type: string; address: string }[],
+) => {
+  const {
+    address,
+    chain: { namespace, reference },
+  } = parseCaipAccountId(account);
+
+  const isSupportedEip155Account = () =>
+    getInternalAccounts().some(
+      (internalAccount) =>
+        ['eip155:eoa', 'eip155:erc4337'].includes(internalAccount.type) &&
+        isEqualCaseInsensitive(address, internalAccount.address),
+    );
+
+  switch (namespace) {
+    case KnownCaipNamespace.Wallet:
+      return reference === KnownCaipNamespace.Eip155
+        ? isSupportedEip155Account()
+        : false;
+    case KnownCaipNamespace.Eip155:
+      return isSupportedEip155Account();
+    default:
+      return false;
+  }
+};
+
+/**
+ * Determines if a method is supported by the wallet.
+ * @param scopeString - The scope string to check.
+ * @param method - The method to check.
+ * @returns A boolean indicating if the method is supported by the wallet.
+ */
+export const isSupportedMethod = (
+  scopeString: ExternalScopeString,
+  method: string,
+): boolean => {
+  const { namespace, reference } = parseScopeString(scopeString);
+
+  if (!namespace || !isKnownCaipNamespace(namespace)) {
+    return false;
+  }
+
+  if (namespace === KnownCaipNamespace.Wallet) {
+    if (reference) {
+      if (
+        !isKnownCaipNamespace(reference) ||
+        reference === KnownCaipNamespace.Wallet
+      ) {
+        return false;
+      }
+      return KnownWalletNamespaceRpcMethods[reference].includes(method);
+    }
+
+    return KnownWalletRpcMethods.includes(method);
+  }
+
+  return KnownRpcMethods[namespace].includes(method);
+};
+
+/**
+ * Determines if a notification is supported by the wallet.
+ * @param scopeString - The scope string to check.
+ * @param notification - The notification to check.
+ * @returns A boolean indicating if the notification is supported by the wallet.
+ */
+export const isSupportedNotification = (
+  scopeString: ExternalScopeString,
+  notification: string,
+): boolean => {
+  const { namespace } = parseScopeString(scopeString);
+
+  if (!namespace || !isKnownCaipNamespace(namespace)) {
+    return false;
+  }
+
+  return (
+    KnownNotifications[namespace as NonWalletKnownCaipNamespace] || []
+  ).includes(notification);
+};
+
+/**
+ * Checks whether the given namespace is a known CAIP namespace.
+ *
+ * @param namespace - The namespace to check
+ * @returns Whether the given namespace is a known CAIP namespace.
+ */
+function isKnownCaipNamespace(
+  namespace: string,
+): namespace is KnownCaipNamespace {
+  const nonWalletKnownCaipNamespaces = Object.keys(KnownCaipNamespace)
+    .filter((key) => key !== KnownCaipNamespace.Wallet)
```

Comment:
My bad, my suggestion had a mistake in it (this was leftover from the first revision, should have been removed in the second):
```suggestion
```

Reply:
In the version of this function that you added, this type guard is just responsible for ensuring the provided namespace is known.

The part where the wallet namespace is filtered out happens separately, in a condition

Reply:
like this https://github.com/MetaMask/core/pull/4784/commits/ba9c728c83cfaeb824f847494b97d17c193dccec ?

Reply:
Oh but yes you're right, an additional check would be needed for the supported notifications function, good catch

Reply:
Sorry for the confusion, this is what I meant: https://github.com/MetaMask/core/pull/4912

Reply:
Make sense. Merged your PR: https://github.com/MetaMask/core/pull/4784/commits/693ee5c47c4de6ad936a12913c45a70ac140fa90